src/components/flat-table/flat-table.pw.tsx

```diff
@@ -0,0 +1,3619 @@
+  });
+
+  test(`should render with zebra stripes`, async ({ mount, page }) => {
+    await mount(<FlatTableComponent isZebra />);
+
+    for (let i = 0; i < 4; i++) {
+      if (i === 0 || i === 2) {
+        await expect(
+          flatTableBodyRowByPosition(page, i).locator("td").nth(0)
+        ).toHaveCSS("background-color", "rgb(255, 255, 255)");
+      } else {
+        await expect(
+          flatTableBodyRowByPosition(page, i).locator("td").nth(0)
+        ).toHaveCSS("background-color", "rgb(250, 251, 251)");
+      }
+    }
+  });
+
+  ([...sizes] as [FlatTableProps["size"], string, string, number][]).forEach(
+    ([sizeName, padding, fontSize, rowHeight]) => {
+      test(`should render as ${sizeName} size`, async ({ mount, page }) => {
+        await mount(<FlatTableComponent size={sizeName} />);
+
+        await assertCssValueIsApproximately(
+          flatTableHeader(page),
+          "height",
+          rowHeight
+        );
+
+        for (let i = 0; i < 4; i++) {
+          const headerCells = flatTableHeaderCells(page)
+            .nth(i)
+            .locator("div")
+            .nth(0);
+          await expect(headerCells).toHaveCSS("padding-left", padding);
+          await expect(headerCells).toHaveCSS("padding-right", padding);
+          await expect(headerCells).toHaveCSS("font-size", fontSize);
+        }
+      });
+    }
+  );
+
+  heightWidth.forEach((height) => {
+    test(`should render with ${height}px as a height parameter`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<FlatTableComponent height={`${height}px`} />);
+
+      await assertCssValueIsApproximately(
+        flatTableWrapper(page),
+        "height",
+        height
+      );
+    });
+  });
+
+  [150, 249, 250, 251, 300].forEach((height) => {
+    test(`should render with ${height}px as a height parameter and minHeight set to 250px`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(
+        <FlatTableComponent height={`${height}px`} minHeight="250px" />
+      );
+
+      if (height < 250) {
+        await expect(flatTableWrapper(page)).toHaveCSS("height", `${250}px`);
+      } else {
+        await assertCssValueIsApproximately(
+          flatTableWrapper(page),
+          "height",
+          height
+        );
+      }
+    });
+  });
+
+  test(`should render with hasMaxHeight parameter`, async ({ mount, page }) => {
+    await mount(<FlatTableComponent height="400px" hasMaxHeight />);
+
+    await expect(flatTableWrapper(page)).toHaveCSS("max-height", "100%");
+  });
+
+  heightWidth.forEach((width) => {
+    test(`should render with ${width}px as a width parameter`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<FlatTableComponent width={`${width}px`} />);
+
+      await assertCssValueIsApproximately(
+        flatTableWrapper(page),
+        "width",
+        width
+      );
+    });
+  });
+
+  ["visible", "hidden", "clip", "scroll", "auto"].forEach((overflow) => {
```

Comment:
suggestion: it might be worth storing this in a variable as it's used again later in the file